Editor/UI/Actions.cs

```diff
@@ -74,6 +74,34 @@ public static void InstallPackage(AvailablePackage package, string version)
 			// TODO: handle errors
 		}
 
+		public static void InstallCustomPackage(string name, string url, string version = null)
+		{
+			// verify this version exists
+			// verify not already installed
+
+			var relativeInstallDirectory = Settings.RelativePackagesDirectoryPath + name;
+			var absoluteInstallDirectory = Settings.AbsolutePackagesDirectoryPath + name;
+			var args = new InstallPackageArgs(name, url, version);
+
+			Dispatcher.Dispatch(ActionTypes.PACKAGE_INSTALLATION_STARTED, args);
+
+			var taskChain = new TaskChain();
+
+			taskChain.Add(
+				new AddSubmoduleTask(url, relativeInstallDirectory));
+
+			if (!string.IsNullOrEmpty(version))
+			{
+				taskChain.Add(new CheckoutSubmoduleTask(absoluteInstallDirectory, version));
+			}
+
+			taskChain.Execute(() =>
```

Comment:
Just an option but I believe you can do:

`taskChain.Execute(() => Dispatcher.Dispatch(ActionTypes.PACKAGE_INSTALLATION_COMPLETE, args));`

Just what you prefer for readability.